### PR TITLE
Fix embeds being cleared

### PR DESCRIPTION
**Pull request by jepcd**
https://discord.com/channels/381880193251409931/381912587505500160/1027735242787794996

### CODE REVIEW

File: src/builder/create_interaction_response.rs [hunk 1]
```diff
@@ -160,13 +163,13 @@ impl CreateInteractionResponseData {
 
     /// Adds an embed to the message.
```
Comment (kangalio) [inline src/builder/create_interaction_response.rs:164]: Should probably add a paragraph like "If you're calling this function while editing a message, existing embeds on the message are cleared" to make it clear what happens to existing embeds (phrasing up for debate)

File: src/builder/create_interaction_response.rs [hunk 2]
```diff
@@ -160,13 +163,13 @@ impl CreateInteractionResponseData {
...
     pub fn add_embed(mut self, embed: CreateEmbed) -> Self {
-        self.embeds.push(embed);
+        self.embeds.get_or_insert_with(Vec::new).push(embed);
         self
     }
 
     /// Adds multiple embeds for the message.
```
Comment (kangalio) [inline src/builder/create_interaction_response.rs:172]: Here too

File: src/builder/create_interaction_response.rs (hunk 1 above)
Comment (jepcd) [reply src/builder/create_interaction_response.rs:164]: But doesn't it not clear the existing embeds?
`.add_embed` and `.add_embeds` just add additional embeds
`.embed` and `.embeds` clear existing embeds, and they already document that

Comment (kangalio) [reply src/builder/create_interaction_response.rs:164]: Say you posted a message with embed A. Editing without calling add_embed() will keep embed A. Editing with calling add_embed(B) will replace the embed with B. Editing with calling add_embed(B).add_embed(C) will replace the embed with B and C

Comment (kangalio) [reply src/builder/create_interaction_response.rs:164]: Discord interprets `None` in the embeds field as "no change" and `Some()` as "replace all existing embeds with these". That's why embeds were previously always cleared; there was no Option so Discord always received Some

Comment (jepcd) [reply src/builder/create_interaction_response.rs:164]: Ah ok I got it, added them

Comment (kangalio) [reply src/builder/create_interaction_response.rs:164]: Feel free to improve the phrasing. It wasn't clear to you so it won't be clear to many others as well

Comment (mkrasnitski) [reply src/builder/create_interaction_response.rs:164]: It's it's best to reword the comment into future tense, and remove the use of 2nd-person. E.g, "calling this while editing..."

Comment (jepcd) [reply src/builder/create_interaction_response.rs:164]: I updated it to 
```rs
/// Calling this while editing a message will overwrite existing embeds.
```

**Review by arqunis — APPROVED** [review]

**Pull request merged**